physx/src/physics.rs

```diff
@@ -736,11 +738,11 @@ impl<Allocator: AllocatorCallback> PhysicsFoundationBuilder<Allocator> {
     /// Set error callback
     /// # Safety
     /// `error_callback` must live as long as the object created by `build`
-    pub unsafe fn with_error_callback(
+    pub unsafe fn with_error_callback<EC: ErrorCallback>(
         &mut self,
-        error_callback: *mut PxErrorCallback,
+        error_callback: EC,
     ) -> &mut Self {
-        self.error_callback = Some(error_callback);
+        self.error_callback = Some(error_callback.into_px());
```

Comment:
so we leak the object here, and the boxed self below.
is that ok?

Reply:
Yes; for now. :-) As noted on Slack we can solve this at a later point without any API changes.